tests/sources/test_confluence.py

```diff
@@ -873,63 +866,59 @@ async def test_fetch_spaces():
     # Setup
     async with create_confluence_source() as source:
         async_response = AsyncMock()
-        async_response.__aenter__ = AsyncMock(
+        async_response.json.return_value = AwaitableAsyncMock(RESPONSE_SPACE)
```

Comment:
Why have an AwaitableAsyncMock here?

We also have `AsyncIterator` helper class, would that help?